src/Dofs/apply_analytical.jl

```diff
@@ -0,0 +1,95 @@
+
+function _default_interpolations(dh::MixedDofHandler)
+    fhs = dh.fieldhandlers
+    getcelltype(i) = typeof(getcells(dh.grid, first(fhs[i].cellset)))
+    ntuple(i -> default_interpolation(getcelltype(i)), length(fhs))
+end
+
+function _default_interpolation(dh::DofHandler)
+    return default_interpolation(typeof(getcells(dh.grid, 1)))
+end
+
+"""
+    apply_analytical!(
+        a::AbstractVector, dh::AbstractDofHandler, fieldname::Symbol, 
+        f::Function, cellset=1:getncells(dh.grid))
+
+Apply a solution `f(x)` by modifying the values in the degree of freedom vector `a`
+pertaining to the field `fieldname` for all cells in `cellset`.
+The function `f(x)` are given the spatial coordinate
+of the degree of freedom. For scalar fields, `f(x)::Number`,
+and for vector fields with dimension `dim`, `f(x)::Vec{dim}`.
+
+This function can be used to apply initial conditions for time dependent problems.
+
+!!! note
+    
+    This function only works for standard nodal finite element interpolations
+    when the function value at the (algebraic) node is equal to the corresponding
+    degree of freedom value.
+    This holds for e.g. Lagrange and Serendipity interpolations, including
+    sub- and superparametric elements.
+"""
+function apply_analytical!(
+    a::AbstractVector, dh::DofHandler, fieldname::Symbol, f::Function,
+    cellset = 1:getncells(dh.grid))
+
+    fieldname ∉ getfieldnames(dh) && error("The fieldname $fieldname was not found in the dof handler")
+    ip_geo = _default_interpolation(dh)
+    field_idx = find_field(dh, fieldname)
+    ip_fun = getfieldinterpolation(dh, field_idx)
+    celldofinds = dof_range(dh, fieldname)
+    field_dim = getfielddim(dh, field_idx)
+    _apply_analytical!(a, dh, celldofinds, field_dim, ip_fun, ip_geo, f, cellset)
+end
+
+function apply_analytical!(
+    a::AbstractVector, dh::MixedDofHandler, fieldname::Symbol, f::Function,
+    cellset = 1:getncells(dh.grid))
+
+    fieldname ∉ getfieldnames(dh) && error("The fieldname $fieldname was not found in the dof handler")
+    ip_geos = _default_interpolations(dh)
+
+    for (fh, ip_geo) in zip(dh.fieldhandlers, ip_geos)
+        fieldname ∈ getfieldnames(fh) || continue
+        field_idx = find_field(fh, fieldname)
+        ip_fun = getfieldinterpolation(fh, field_idx)
+        field_dim = getfielddim(fh, field_idx)
+        celldofinds = dof_range(fh, fieldname)
+        _apply_analytical!(a, dh, celldofinds, field_dim, ip_fun, ip_geo, f, intersect(fh.cellset, cellset))
+    end
+    return a
+end
+
+function _apply_analytical!(
+    a::Vector, dh::AbstractDofHandler, celldofinds, field_dim,
+    ip_fun::Interpolation, ip_geo::Interpolation, f::Function, cellset)
+
+    coords = getcoordinates(dh.grid, first(cellset))
+    cdv = CellDofValues(ip_fun, ip_geo)
```

Comment:
Seems like this could simply be `CellScalarValues` with a custom quadrature rule with points in the reference coordinates?

Reply:
OK, I'll look into that when I have time. 
I guess this should be a form of 1st order lobatto quadrature then. But since this is not implemented for RefTet I have to think a bit how to fix that. 
Will ping once fixed!

Reply:
You can just plug whatever coordinates you want in a "fake" quadrature rule, and think of them as "points for evaluation" instead of quadrature points, for example
```
julia> using Ferrite

julia> ip = Lagrange{2,RefTetrahedron,1}();

julia> points_for_evaluation = Ferrite.reference_coordinates(ip);

julia> fake_quadrature_rule = QuadratureRule{2,RefTetrahedron,Float64}(fill(NaN, 3), points_for_evaluation);

julia> cellvalues = CellScalarValues(fake_quadrature_rule, ip)
CellScalarValues{2, Float64, RefTetrahedron} with 3 shape functions and 3 quadrature points
```

You can add a constructor like
```
function QuadratureRule{D,R}(weights::AbstractVector{T}, points::AbstractVector{Vec{D,T}}) where {D, R, T}
    QuadratureRule{D,R,T}(weights, points)
end
```
to make it easier (no need to specify `T`) (see also https://github.com/Ferrite-FEM/Ferrite.jl/issues/513).

Reply:
Thanks for those tips - now I also learned how to do custom quad rules!
I reduced away the dim input in the constructor as well - hope that is ok!